internal/service/advancedclustertpf/resource_compatibility_reuse.go

```diff
@@ -17,6 +18,54 @@ func FormatMongoDBMajorVersion(version string) string {
 	return fmt.Sprintf("%.1f", cast.ToFloat32(version))
 }
 
+func resolveExtraAPIInfo(ctx context.Context, projectID string, cluster *admin.ClusterDescription20240805, api admin.NetworkPeeringApi) (*ExtraAPIInfo, error) {
+	containerIDs := map[string]string{}
+	for _, spec := range cluster.GetReplicationSpecs() {
+		for _, regionConfig := range spec.GetRegionConfigs() {
+			providerName := regionConfig.GetProviderName()
+			if providerName == constant.TENANT {
+				continue
+			}
+			params := &admin.ListPeeringContainerByCloudProviderApiParams{
+				GroupId:      projectID,
+				ProviderName: &providerName,
+			}
+			containerIDKey := fmt.Sprintf("%s:%s", providerName, regionConfig.GetRegionName())
+			if _, ok := containerIDs[containerIDKey]; ok {
+				continue
+			}
+			containers, _, err := api.ListPeeringContainerByCloudProviderWithParams(ctx, params).Execute()
+			if err != nil {
+				return nil, err
+			}
+			if results := getAdvancedClusterContainerID(containers.GetResults(), &regionConfig); results != "" {
+				containerIDs[containerIDKey] = results
+			}
+		}
+	}
+	return &ExtraAPIInfo{
+		ContainerIDs: containerIDs,
+	}, nil
+}
+
+// copied from model_advanced_cluster.go
+func getAdvancedClusterContainerID(containers []admin.CloudProviderContainer, cluster *admin.CloudRegionConfig20240805) string {
+	if len(containers) == 0 {
+		return ""
+	}
+	for i := range containers {
+		if cluster.GetProviderName() == constant.GCP {
+			return containers[i].GetId()
+		}
+		if containers[i].GetProviderName() == cluster.GetProviderName() &&
+			containers[i].GetRegion() == cluster.GetRegionName() || // For Azure
+			containers[i].GetRegionName() == cluster.GetRegionName() { // For AWS
+			return containers[i].GetId()
+		}
```

Comment:
```suggestion
                gpc := cluster.GetProviderName() == constant.GCP
                azure := containers[i].GetProviderName() == cluster.GetProviderName() && containers[i].GetRegion() == cluster.GetRegionName() 
                aws := containers[i].GetRegionName() == cluster.GetRegionName()

		if check gpc || azure || aws {
			return containers[i].GetId()
		}
```

I like to try to express doc directly in the code.

I'm assuming that condition is fine and `containers[i].GetProviderName() == cluster.GetProviderName() &&` is only for Azure as && has precedence over || so it's done before.    But maybe it's wrong and we want to do ` containers[i].GetProviderName() == cluster.GetProviderName()` also for AWS.